### PR TITLE
[Fix] Use RECOGNIZER instead of DETECTORS

### LINKED ISSUE #595: RECOGNIZERS Registry Code Design
I find the Registry RECOGNIZERS defined in mmocr/models/builder.py never be used in somewhere else, and the "recognizer" model like SARNet was registered in DETECTOR Register. Is here some reason about this code design?

### PR DESCRIPTION
**Pull request by cuhk-hbsun**
fix #595

### CODE REVIEW

File: mmocr/models/builder.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 HEADS = BACKBONES
 NECKS = BACKBONES
 FUSERS = BACKBONES
+RECOGNIZERS = BACKBONES
```
Comment (gaotongxiao) [inline mmocr/models/builder.py:25]: Remove line 10

**Comment by gaotongxiao** [comment]
BTW, it's not a breaking change since RECOGNIZERS is an alias of `model` Registry @Harold-lkk

**Comment by Harold-lkk** [comment]
@gaotongxiao  actually it is. Using RECOGNIZERS need to build the model with build_recognizer,  which effect a lot, like train, test, inference, torchserve and depoly

**Comment by gaotongxiao** [comment]
@Harold-lkk `build_detector` still works implicitly. Its name may arise some confusion though. @cuhk-hbsun may need to rename `init_detector`, `build_detector` to more general names like `init_models` and``build_models`.

**Comment by Harold-lkk** [comment]
build_detector will not work, because ` DETECTORS.build(
        cfg, default_args=dict(train_cfg=train_cfg, test_cfg=test_cfg))`. Also if use build model, need to get a task name, which actually effect a lot which may need a doc for designing

**Comment by gaotongxiao** [comment]
@Harold-lkk I didn't get your point. Anyway, recognizers have passed the test build.
https://github.com/open-mmlab/mmocr/blob/0a1787d6bcbec25649441a1f26e928ea7158e4c4/tests/test_models/test_recog_config.py#L108-L119

**Comment by gaotongxiao** [comment]
Need a follow-up PR that renames init_detector before we merge this one. Otherwise, it would become rather confusing.

**Review by gaotongxiao — APPROVED** [review]
Need a follow-up PR that renames init_detector before we merge this one. Otherwise, it would become rather confusing.

**Pull request merged**